presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -104,6 +112,13 @@ static int calculateBlockResetBytes(int currentBytes)
         return newOffsets;
     }
 
+    /**
+     * Returns the slice starting at specified index and extends for the specified length
+     * in a compact representation.
+     * <p>
+     * If the whole slice is requested and the input slice is already in a compact representation,
+     * the input slice will be returned.
```

Comment:
```
Returns a slice containing values in the specified range of the specified slice.
If the range matches the entire slice, the input slice will be returned.
Otherwise, a copy will be returned.